src/solver/optim-model-filler/TimeDependentLinearExpression.cpp

```diff
@@ -150,8 +150,9 @@ TimeDependentLinearExpression TimeDependentLinearExpression::shiftLinearExpressi
     LinearExpressionMap linearExpressions;
     for (const auto& timeStep: linearExpressions_ | std::views::keys)
     {
-        linearExpressions[timeStep] = linearExpressions_.at(
-          rotatedIndex(timeStep, shiftValue, fillContext));
+        linearExpressions.emplace(timeStep,
+                                  linearExpressions_.at(
+                                    rotatedIndex(timeStep, shiftValue, fillContext)));
     }
```

Comment:
`linearExpression` can be mutated in the loop

```suggestion
for (auto& [timeStep, linearExpression]: linearExpressions_)
{
        linearExpression = linearExpressions_.at(rotatedIndex(timeStep, shiftValue, fillContext)));
}
```

Reply:
no it can't. if we want to do:
t  = 1 -> t = 3
t = 4 -> t = 1 (if t=1 is modified then the logic is broken )

Reply:
agreed